libultraship/libultraship/Archive.cpp

```diff
@@ -320,8 +323,64 @@ namespace Ship {
 		return true;
 	}
 
+    void Archive::GenerateCRCMap() {
+        auto listFile = LoadFile("(listfile)", false);
+
+        std::vector<std::string> lines = StringHelper::Split(std::string(listFile->buffer.get(), listFile->dwBufferSize), "\n");
+
+        for (size_t i = 0; i < lines.size(); i++)
+        {
+            std::string line = StringHelper::Replace(StringHelper::Strip(lines[i], "\r"), "/", "\\");
+            std::string line2 = StringHelper::Replace(line, "\\", "/");
+
+            uint64_t hash = CRC64(StringHelper::Replace(line, "/", "\\").c_str());
+            uint64_t hash2 = CRC64(StringHelper::Replace(line, "\\", "/").c_str());
+            hashes[hash] = line;
+            hashes[hash2] = line2;
+        }
+    }
+
+    void Archive::PushGameVersion() {
+        auto t = LoadFile("version", false);
+        if (!t->bHasLoadError)
+        {
+            uint32_t gameVersion = (*((uint32_t *)t->buffer.get()));
```

Comment:
This should probably use a binary reader to get the data so that we can have endian independence with this information.

Reply:
OK, I wasn't familiar with how we normally got the version so I copied that from somewhere else. I should be able to use the BinaryReader in the ResourceManager yeah? Or at least a similar implementation? I'll look into it.

Reply:
Actually I guess the BinaryReader is part of ZAPD, but I should be able to import the header file like ResourceMgr does.

Reply:
Hmm, trying to use the Binary Reader already in ZAPD, but now it won't build on linux due to undefined references, but I'm not sure why since other files in libultraship are already using it just fine.

Reply:
Ok, got that BinaryReader that will work implemented correctly, but I also discovered that the version file doesn't have a header with it's Endianness in it, so I will need to add that before this is truly complete.

Reply:
Ok, version file has a byte representing it's Endianness before the actual version now.